### PR TITLE
refactor: update `blas/ext/base/dapxsumkbn2` to follow current project conventions

### LINKED ISSUE #1467: [RFC]: refactor `blas/ext/base/dapxsumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dapxsumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dapxsumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by kumarsuraj212003**
Resolves #1467 .

## Description

> What is the purpose of this pull request?

refactor `blas/ext/base/dapxsumkbn2` to follow current project conventions

This pull request:

## Related Issues
#788, #1152

> Does this pull request have any related issues?

This pull request:

-   resolves #1467
-   progresses #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/manifest.json [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
+    "options": {},
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/manifest.json]: Can you please change this file to 2-space indentation and add configurations for the examples and benchmarks?

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/src/addon.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+    STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+    STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+    STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
+    STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+    STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, stride, argv, 2 );
+
+    napi_value v;
+    napi_status status = napi_create_double( env, stdlib_strided_dapxsumkbn2( N, alpha, (double *)X, stride ), &v );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/src/addon.c]: The cast should not be needed anymore: 

```suggestion
    napi_status status = napi_create_double( env, stdlib_strided_dapxsumkbn2( N, alpha, X, stride ), &v );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/manifest.json [hunk 3]
```diff
@@ -1,40 +1,45 @@
...
+            ],
+            "libraries": [
+                "-lm"
+            ],
+            "libpath": [],
+            "dependencies": [
+                "stdlib/napi/export.h",
+                "stdlib/napi/argv.h",
+                "stdlib/napi/argv_int64.h",
+                "stdlib/napi/argv_strided_float64array.h"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/manifest.json]: The dependencies should be listed as stdlib packages:

```suggestion
                "@stdlib/napi/export",
                "@stdlib/napi/argv",
                "@stdlib/napi/argv-int64",
                "@stdlib/napi/argv-strided-float64array"
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for working on this! Left a few comments; mainly the `manifest.json` file has to be updated. Please have a look at how we have been updating it in some of the linked PRs in the RFC.

**Comment by kumarsuraj212003** [comment]
@Planeshifter thanks for review, I added suggested changes, can you please tell me why my  `run_affected_tests / Run affected tests (pull_request)` test is failling or Do I need to update any particular dependencies?

**Comment by kgryte** [comment]
@kumarsuraj212003 The `manifest.json` file still needs updating.

**Comment by kumarsuraj212003** [comment]
@kgryte `run_affected_tests / Run affected tests (pull_request)` is still failing. Could you please investigate the error.It helps me alot. Thank you

**Comment by kgryte** [comment]
Done. You're missing an include and the `manifest.json` file is missing a dep.

**Comment by kumarsuraj212003** [comment]
@kgryte I think I understood what you wanted me to correct, and I have made the necessary adjustments, including adding the missing include and updating the dependencies in the `manifest.json` file. Thank you again for helping me identify and resolve my errors.

**Review by Planeshifter — APPROVED** [review]
Thanks for getting this PR into a good shape! LGTM. I will merge it shortly.

**Pull request merged**